app/controllers/staff/reports_state_controller.rb

```diff
@@ -24,89 +32,40 @@ def edit
+      change_report_state_to(:submitted)
     else
       authorize report
       redirect_to report_path(report)
     end
   end
 
-  private def confirm_activation
-    authorize report, :activate?
-    @report_presenter = ReportPresenter.new(report)
-    render "staff/reports_state/activate/confirm"
-  end
-
-  private def change_report_state_to_active
-    authorize report, :activate?
-    report.update!(state: :active)
-    report.create_activity key: "report.activated", owner: current_user
-    @report_presenter = ReportPresenter.new(report)
-    render "staff/reports_state/activate/complete"
-  end
-
-  private def confirm_submission
-    authorize report, :submit?
-    @report_presenter = ReportPresenter.new(report)
-    render "staff/reports_state/submit/confirm"
-  end
-
-  private def change_report_state_to_submitted
-    authorize report, :submit?
-    report.update!(state: :submitted)
-    report.create_activity key: "report.submitted", owner: current_user
-    @report_presenter = ReportPresenter.new(report)
-    render "staff/reports_state/submit/complete"
-  end
-
-  private def confirm_in_review
-    authorize report, :review?
-    @report_presenter = ReportPresenter.new(report)
-    render "staff/reports_state/review/confirm"
-  end
-
-  private def change_report_state_to_in_review
-    authorize report, :review?
-    report.update!(state: :in_review)
-    report.create_activity key: "report.in_review", owner: current_user
-    @report_presenter = ReportPresenter.new(report)
-    render "staff/reports_state/review/complete"
-  end
-
-  private def confirm_request_changes
-    authorize report, :request_changes?
-    @report_presenter = ReportPresenter.new(report)
-    render "staff/reports_state/request_changes/confirm"
-  end
-
-  private def change_report_state_to_awaiting_changes
-    authorize report, :request_changes?
-    report.update!(state: :awaiting_changes)
-    report.create_activity key: "report.awaiting_changes", owner: current_user
+  private def confirm_state(state)
+    authorize report, state + "?"
     @report_presenter = ReportPresenter.new(report)
-    render "staff/reports_state/request_changes/complete"
+    render "staff/reports_state/#{state}/confirm"
   end
 
-  private def confirm_approve
-    authorize report, :approve?
-    @report_presenter = ReportPresenter.new(report)
-    render "staff/reports_state/approve/confirm"
-  end
+  private def change_report_state_to(state)
+    policy = STATE_TO_POLICY.fetch(state).to_s
 
-  private def change_report_state_to_approved
-    authorize report, :approve?
-    report.update!(state: :approved)
-    report.create_activity key: "report.approve", owner: current_user
-    @report_presenter = ReportPresenter.new(report)
-    render "staff/reports_state/approve/complete"
+    authorize report, policy + "?"
+    if report.valid?
+      report.update!(state: state)
+      report.create_activity key: "report.state.changed_to.#{state}", owner: current_user
```

Comment:
We've changed the names of the events sent to the audit log here -- is that a problem or is it fine to change those?